components/module-components/SearchAndFilter/DisplayNames.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useEffect } from 'react';
+import submitSearchForm from './SubmitSearchForm';
+import style from './displayNames.module.scss';
+import { UserRole } from './Filter';
+
+interface Author {
+  id: string;
+  first_name: string;
+  last_name: string;
+  role: string;
+}
+
+interface DisplayNamesProps {
+  filterValue: string;
+  selectedRoles: UserRole[];
+}
+
+const DisplayNames = ({ filterValue, selectedRoles }: DisplayNamesProps) => {
+  const [names, setNames] = useState<Author[]>([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await submitSearchForm();
+        const filteredNames = response.filter((author: Author) => {
+          const fullName = `${author.first_name} ${author.last_name}`.toLowerCase();
+          return (
+            fullName.includes(filterValue.toLowerCase()) &&
+            selectedRoles.includes(author.role as UserRole)
+          );
+        });
+        setNames(filteredNames);
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+
+    fetchData();
+  }, [filterValue, selectedRoles]);
```

Comment:
better error handling here, please use try catch

Reply:
Not sure what to do,its already try catch